File: usvm-jvm-spring/usvm-jvm-spring-api/src/main/java/org/usvm/spring/query/specification/SortSpecification.java
package org.usvm.spring.query.specification;

public class SortSpecification {
    public ASpecification spec;
    public boolean isAscending = true; // true -  ASC, false - DESC
    public boolean isNullsLast = true; // true - LAST, false - FIRST

    public SortSpecification() {
    }

    public SortSpecification(ASpecification spec, boolean isAscending, boolean isNullsLast) {
        this.spec = spec;
        this.isAscending = isAscending;
        this.isNullsLast = isNullsLast;
    }
}

File: usvm-jvm-spring/src/main/kotlin/machine/interpreter/transformers/springjpa/query/specification/SortSpecification.kt
package machine.interpreter.transformers.springjpa.query.specification

import machine.interpreter.transformers.springjpa.query.CommonInfo
import machine.interpreter.transformers.springjpa.query.MethodCtx
import org.usvm.spring.query.specification.SortSpecification

fun SortSpecification.getLambdas(info: CommonInfo) = spec.getLambdas(info)
fun SortSpecification.getTranslate(ctx: MethodCtx) = spec.getTranslate(ctx)
fun SortSpecification.getTranslateRetType(ctx: MethodCtx) = spec.getTranslateRetType(ctx)
fun SortSpecification.getComparer(ctx: MethodCtx) = spec.getComparer(ctx)
